devops/jobs/SAMLSSLExpirationCheck.groovy

```diff
@@ -0,0 +1,150 @@
+/*
+ Variables without defaults are marked (required) 
+ 
+ Variables consumed for this job:
+    * DEPLOYMENTS (required)
+         deployment:
+            environments:
+              environment (required)
+                 saml_cert_file (required)
+    * FROM_ADDRESS: email address to send notifications that certificates are expiring
+    * TO_ADDRESS: email address that sends the mail
+    * SECURE_GIT_CREDENTIALS: secure-bot-user (required)
+    * SYSADMIN_REPO: repository containing SSL expiration check python script (required)
+    * SYSADMIN_BRANCH: default is master
+    * CONFIGURATION_REPO: name of config repo, default is https://github.com/edx/configuration.git
+    * CONFIGURATION_BRANCH: default is master
+    * CONFIGURATION_SECURE_REPO: name of config secure repo
+    * CONFIGURATION_SECURE_BRANCH: default is master
+    * REGION: default is us-east-1
+    * NOTIFY_ON_FAILURE: alert@example.com
+    * FOLDER_NAME: folder, default is Monitoring
+    * DAYS: alert if SSL certificate will expire within these days
+    
+
+ This job expects the following credentials to be defined on the folder
+    tools-edx-jenkins-aws-credentials: file with key/secret in boto config format
+    saml-ssl-expiration-check-${deployment}-role-arn: the role to aws sts assume-role
+
+*/
+
+package devops.jobs
+import static org.edx.jenkins.dsl.Constants.common_wrappers
+import static org.edx.jenkins.dsl.Constants.common_logrotator
+
+class SAMLSSLExpirationCheck{
+    public static def job = { dslFactory, extraVars ->
+        assert extraVars.containsKey('DEPLOYMENTS') : "Please define DEPLOYMENTS. It should be a list of strings."
+        assert !(extraVars.get('DEPLOYMENTS') instanceof String) : "Make sure DEPLOYMENTS is a list and not a string"
+        extraVars.get('DEPLOYMENTS').each { deployment, configuration ->
+                configuration.environments.each { environment, inner_config -> 
+                    dslFactory.job(extraVars.get("FOLDER_NAME","Monitoring") + "/saml-ssl-expiration-check-${environment}-${deployment}") {
+                        wrappers common_wrappers
+                        logRotator common_logrotator
+
+                        wrappers {
+                            credentialsBinding {
+                            file('AWS_CONFIG_FILE','tools-edx-jenkins-aws-credentials')
+                            string('ROLE_ARN', "saml-ssl-expiration-check-${deployment}-role-arn")
+                            }
+                        }
+
+                        def gitCredentialId = extraVars.get('SECURE_GIT_CREDENTIALS','')
+                        def config_secure_repo = "git@github.com:edx-ops/${deployment}-secure.git"
```

Comment:
Let's pass the whole repo path from config instead of requiring edx-ops specific repos.